### DISCUSSION TITLE
Form examples

### DISCUSSION THREAD

**Discussion by tengl** [GregoryConrad/rearch-dart#228] [General]
Interesting project! Will try it out. 

Can you make an example with form data? I've done some tests but it don't really like how it turned out. So I'd like so see some examples by someone who knows how it should be done. 

It would be great if the form would have different inputs, like; text, number, time picker, check boxes and a send button at least. Also a validation of required fields and perhaps a clear form button.

**Comment by GregoryConrad** [comment]
Hi! 👋

I'm super busy right now so a full example may not come for a long time, but here's something fast to get you started:

```dart
SomeWidget(
  child: Form(
    child: RearchBuilder(
      builder: (context, use) {
        final form = Form.of(context); // so you can validate, reset, etc.
        final textController = use.textEditingController();
        final (selected, setSelected) = use.state(false);
        return Placeholder(); // TODO use above variables to make your Form UI, feel free to make others as needed
      },
    ),
  ),
);
```

**Reply by GregoryConrad** [reply]
(You can also make a `RearchForm[Builder]` class to wrap around the `Form` + `RearchBuilder` combo if you use it a few times)
